qiskit_experiments/randomized_benchmarking/interleaved_rb_analysis.py

```diff
@@ -0,0 +1,167 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Interleaved RB analysis class.
+"""
+from typing import Optional, List
+import numpy as np
+from qiskit_experiments.analysis.curve_fitting import (
+    process_multi_curve_data,
+    multi_curve_fit,
+)
+
+from qiskit_experiments.analysis.plotting import plot_curve_fit, plot_scatter, plot_errorbar
+from qiskit_experiments.analysis.data_processing import (
+    level2_probability,
+    multi_mean_xy_data,
+)
+from .rb_analysis import RBAnalysis
+
+try:
+    from matplotlib import pyplot as plt
+
+    HAS_MATPLOTLIB = True
+except ImportError:
+    HAS_MATPLOTLIB = False
+
+
+class InterleavedRBAnalysis(RBAnalysis):
+    """Interleaved RB Analysis class.
+    According to the paper: "Efficient measurement of quantum gate
+    error by interleaved randomized benchmarking" (arXiv:1203.4550)
+    - Equations (4) and (5).
```

Comment:
Its not very useful to mention paper equation number in the doc string. Ideally it should be rewritten to include a minimal description of the relevant fitter equations in the doc string, and include a hyperlink reference to the PRL/Arxiv papers as references for people who want more details.